package.json

```diff
@@ -364,6 +364,11 @@
           "default": false,
           "description": "Suppress the notification window to perform a 'dotnet restore' when dependencies can't be resolved."
         },
+        "csharp.suppressHiddenMessages": {
+            "type": "boolean",
+            "default": true,
+            "description": "Suppress hidden messages from being shown."
```

Comment:
How about we go with "hidden diagnostics" rather than "hidden messages".

* setting name: `csharp.suppressHiddenDiagnostics`
* description: `Suppress 'hidden' diagnostics (such as 'unnecessary using directives') from appearing in the editor or the Problems pane`